### DISCUSSION TITLE
Gnosis RPC provided by wallet is slow and unresponsive when connected to gnosis-safe.

### DISCUSSION THREAD

**Discussion by mohandast52** [WalletConnect/walletconnect-monorepo#2813] [Developer Support]
Explanation:
* We recently integrated wallet-connect v2 and tested it successfully with Goerli network ⇔ gnosis-safe, where we were able to get the transaction modal, receive responses from contract calls, etc - All good
* However, we are currently experiencing a problem connecting from the gnosis network ⇔ gnosis-safe. The issue is that I cannot receive contract call responses due to the slow RPC. We also used a provider called [JsonProvider](https://wagmi.sh/react/providers/jsonRpc) to pass the RPC URLs, but it is not functioning as expected.

This is what the config looks like now:
```js
export const rpc = {
  1: process.env.NEXT_PUBLIC_MAINNET_URL,
  5: process.env.NEXT_PUBLIC_GOERLI_URL,
  100: process.env.NEXT_PUBLIC_GNOSIS_URL,
  137: process.env.NEXT_PUBLIC_POLYGON_URL,
  31337: process.env.NEXT_PUBLIC_AUTONOLAS_URL,
};

const { publicClient, webSocketPublicClient, chains } = configureChains(
  [
    mainnet,
    goerli,
    gnosis,
    gnosisChiado,
    polygon,
  ],
  [
    w3mProvider({ projectId: process.env.NEXT_PUBLIC_WALLET_PROJECT_ID }),
    jsonRpcProvider({
      rpc: (chain) => {
        return {
          http: rpc[chain.id],
        };
      },
    }),
  ],
);

export const wagmiConfig = createConfig({
  autoConnect: true,
  logger: {
    warn: null,
  },
  connectors: [
    ...w3mConnectors({
      projectId,
      version: 2, // v2 of wallet connect
      chains,
    }),
    new SafeConnector({
      chains,
      options: {
        allowedDomains: [/gnosis-safe.io$/, /app.safe.global$/],
        debug: false,
      },
    }),
  ],
  publicClient,
  webSocketPublicClient,
});

export const ethereumClient = new EthereumClient(wagmiConfig, chains);
```

Any help would be really helpful, thanks!

**Comment by crypblizz8** [comment]
Just to confirm it is the isolation with Gnosis that is slow? 

What were the results on the other RPC networks?
